rules-reviewed/eap7/eap6/hsearch.windup.xml

```diff
@@ -0,0 +1,880 @@
+                        title="Miscelanous Changes in Hibernate Search 5.x" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hsearch-00112">
+            <when>
+                <javaclass references="org.hibernate.search.event.impl.FullTextIndexEventListener">
+                    <location>INHERITANCE</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="FullTextIndexEventListener class is now final in Hibernate Search 5" effort="1" severity="mandatory">
+                    <message>The class `FullTextIndexEventListener` is now a final class. If you currently extend this class, you must find an alternate solution to achieve the same functionality. You could in some cases use an `EntityIndexingInterceptor`.</message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_miscellaneous_hibernate_search_changes"
+                        title="Miscelanous Changes in Hibernate Search 5.x" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hsearch-00113">
+            <when>
+                <javaclass references="org.hibernate.search.backend.impl.jms.AbstractJMSHibernateSearchController">
+                    <location>INHERITANCE</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Usage of internal AbstractJMSHibernateSearchController class in Hibernate Search 5" effort="1"
+                    severity="mandatory">
+                    <message>The JMS backend was depending to Hibernate ORM. This dependency was removed, so that the backend can be used in other (non ORM) environments as well. 
+                    A consequence is that implementors of `org.hibernate.search.backend.impl.jms.AbstractJMSHibernateSearchController` will need to adjust to the new signature.
+                     This class is really considered internal. We suggest to take this class as an example instead of extending it.</message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_miscellaneous_hibernate_search_changes"
+                        title="Miscelanous Changes in Hibernate Search 5.x" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hsearch-00114">
+            <when>
+                <javaclass references="org.hibernate.search.engine.service.spi.ServiceProvider">
+                <!-- it doesn't catch the implements in test code -->
+                    <location>IMPLEMENTS_TYPE</location>
```

Comment:
not working, there is https://github.com/mareknovotny/windup-rulesets/blob/d0b82c2d4a5c3903ffcb0f8335a43f2fdd5f6cdd/rules-reviewed/eap7/eap6/tests/data/data-hsearch/CacheServiceProvider.java#L6